### DISCUSSION TITLE
Default initialization of tree_sitter::Node

### DISCUSSION THREAD

**Discussion by bhargavah** [tree-sitter/tree-sitter#3507] [Q&A] [answered]
Is there any way to initialize a default value to tree_sitter::Node? So, this is how we parse using tree_sitter.

```
let mut parser = tree_sitter::Parser::new();
let path = "abc.txt";
let contents = std::fs::read_to_string(&path).unwrap();
let tree : tree_sitter::Tree = parser.parse(contents, None).unwrap();
let cursor : tree_sitter::TreeCursor = &mut tree.walk();
```

This is fine and we can write an loop to walk over the tree and take actions on the nodes. I was wondering if there is any way to default initialize the Node. I mean declare an global variable or so and fill it later.

Something like:

`let mut prev_node : tree_sitter::Node = tree_sitter::Node();`

OR

`let mut prev_node : tree_sitter::Node = new tree_sitter::Node();`

**Comment by amaanq** [comment, marked as answer]
No, there is no default value because that makes *no* sense to implement from the API (what happens if you call any method on a "default" node?). You should use an Option instead.
